File: frontend/src/microservices/order/objects/RefundPlannerResponse.ts
// 本文件定义订单退款响应，和后端 `RefundPlannerResponse` 保持同名镜像。

export type RefundPlannerResponse = {
  refundId: string
  refundAmount: string
  refundCurrency: string
  refundReason: string
  refundStatus: string
  requestedAt: string
  approvedAt: string | null
  settledAt: string | null
}

export const refundPlannerResponseFromJson = (json: string): RefundPlannerResponse =>
  JSON.parse(json) as RefundPlannerResponse

export const refundPlannerResponseToJson = (value: RefundPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/order/objects/RefundPlannerResponse.scala
package com.typesafe.travel.order.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class RefundPlannerResponse(refundId: String, refundAmount: String, refundCurrency: String, refundReason: String, refundStatus: String, requestedAt: String, approvedAt: Option[String], settledAt: Option[String])
object RefundPlannerResponse:
  given Encoder[RefundPlannerResponse] = deriveEncoder
  given Decoder[RefundPlannerResponse] = deriveDecoder
